packages/pong-engine/src/queue.dto.ts

```diff
@@ -0,0 +1,48 @@
+export type GameChallengeDto = {
+  gameRoomId: string;
+  from: {
+    username: string;
+    id: string;
+  };
+};
+
+export enum GameChallengeStatus {
+  CHALLENGE_DECLINED = 'challengeDeclined',
+  CHALLENGE_ACCEPTED = 'challengeAccepted',
+}
+
+export enum GameStatus {
+  READY = 'ready',
+  FINISHED = 'finished',
+}
+
+export type GameChallengeResponseDto = {
+  gameRoomId: string;
+  status: GameChallengeStatus;
+};
+
+type GameAcceptedDto = {
+  status: GameChallengeStatus.CHALLENGE_ACCEPTED;
+  gameRoomId: string;
+};
+
+type GameReadyDto = {
+  status: GameStatus.READY;
+  gameRoomId: string;
+};
+
+type GameDeclinedDto = {
+  status: GameChallengeStatus.CHALLENGE_DECLINED;
+  gameRoomId?: never;
+};
+
+type GameFinishedDto = {
+  status: GameStatus.FINISHED;
+  gameRoomId?: never;
+};
```

Comment:
It will complain that I'm trying to set a string|undefined to a string|null for gameroom id. And like this we pack all the semantics I'm one piece and can be very specific.